src/jormungandr/jormungandr/src/network/subscription.rs

```diff
@@ -288,6 +245,338 @@ impl Sink<net_data::Header> for BlockAnnouncementProcessor {
+                                }
+                                _ => {
+                                    // In this case the whitelisted multiaddr is invalid
+                                    // so consider it as not matching anything
+                                    tracing::error!(multiaddr = %ma, "Invalid entry in whitelist");
+                                    false
+                                }
+                            }
+                        })
+                    }
+                    None => {
+                        tracing::debug!("no whitelist configured, bypassing check");
+
+                        true
+                    }
+                };
+
+                if should_send {
+                    let fragments = self.clear_buffered_fragments();
+
+                    let addr = match self.global_state.config.address() {
+                        Some(addr) => FragmentOrigin::Network { addr: addr.ip() },
+                        None => {
+                            tracing::info!(
+                                "node addr not present in config, reverting to local lookup"
+                            );
+                            FragmentOrigin::Network {
+                                addr: retrieve_local_ip(),
+                            }
+                        }
+                    };
+
+                    let (reply_handle, _reply_future) = intercom::unary_reply();
+                    self.mbox
+                        .start_send(TransactionMsg::SendTransactions {
+                            origin: addr,
+                            fragments,
+                            fail_fast: false,
+                            reply_handle,
+                        })
+                        .map_err(|e| {
+                            tracing::error!(
+                                reason = %e,
+                                "failed to send fragments to the fragment task"
+                            );
+                            Error::new(Code::Internal, e)
+                        })?;
+                    self.refresh_stat();
+
+                    tracing::debug!("processed fragments from peer");
+
+                    (
+                        FragmentProcessorSendFragmentsState::WaitingMessageBox,
+                        Poll::Ready(Ok(())),
+                    )
+                } else {
+                    self.clear_buffered_fragments();
+
+                    tracing::info!(address = %ingress_addr, "dropping fragments from peer");
+
+                    (
+                        FragmentProcessorSendFragmentsState::WaitingMessageBox,
+                        Poll::Ready(Ok(())),
+                    )
+                }
+            }
+        };
+
+        self.send_fragments_state = next_st;
+        ret
+    }
+
+    fn poll_flush_mbox(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Error>> {
```

Comment:
when does this happen? wondering if there any implications as the mbox is shared among nodes right?